contracts/coordinator/src/state.rs

```diff
@@ -243,23 +243,24 @@ pub fn is_prover_registered(
     storage: &dyn Storage,
     prover_address: ProverAddress,
 ) -> Result<bool, Error> {
-    Ok(CHAIN_PROVER_INDEXED_MAP
-        .idx
-        .by_prover
-        .item(storage, prover_address)
-        .change_context(Error::StateParseFailed)?
-        .is_some())
+    match contracts_by_prover(storage, prover_address) {
+        Ok(..) => Ok(true),
+        Err(e)
+            if e.downcast_ref::<Error>()
+                .is_some_and(|e| matches!(e, Error::ProverNotRegistered(..))) =>
+        {
+            Ok(false)
+        }
+        Err(e) => Err(e),
+    }
 }
 
 #[allow(dead_code)] // Used in tests, might be useful in future query
 pub fn load_prover_by_chain(
     storage: &dyn Storage,
     chain_name: ChainName,
 ) -> Result<ProverAddress, Error> {
-    Ok(CHAIN_PROVER_INDEXED_MAP
-        .may_load(storage, chain_name.clone())
-        .change_context(Error::StateParseFailed)?
-        .ok_or(Error::ChainNotRegistered(chain_name))?)
+    Ok(contracts_by_chain(storage, chain_name)?.prover_address)
 }
```

Comment:
get rid of the whole function. I'm not a fan of building test specific code into production. Sometimes it's necessary, but getting the prover from all chain contracts is now trivial